shell/platform/embedder/embedder_layers.cc

```diff
@@ -164,10 +150,22 @@ void EmbedderLayers::PushPlatformViewLayer(
       }
     }
 
-    if (mutations_array.size() > 0) {
+    if (!mutations_array.empty()) {
+      // If there are going to be any mutations, they must first take into
+      // account the transformation for the device pixel ratio and root surface
+      // transformation.
+      auto base_xformation =
+          SkMatrix::Concat(root_surface_transformation_,
+                           SkMatrix::MakeScale(device_pixel_ratio_));
```

Comment:
I realize this is unrelated to the issue you are addressing, but I don't think the `device_pixel_ratio_` correction should be done here. The framework does this explicitly by pushing a transform layer onto the  scene. When this was initially written, I made a mistake in understanding framework behavior. I can patch this separately.

Reply:
Oh, interesting. Let me test. I'll likely want to fix this now rather than wait.

Reply:
Done and fixed the unittests.